python/cudf/tests/test_index.py

```diff
@@ -98,3 +101,41 @@ def test_index_immutable():
     gi = GenericIndex(np.arange(start, stop))
     with pytest.raises(TypeError):
         gi[1] = 5
+
+
+def test_categorical_index():
+    pdf = pd.DataFrame()
+    pdf['a'] = [1, 2, 3]
+    pdf['index'] = pd.Categorical(['a', 'b', 'c'])
+    pdf = pdf.set_index('index')
+    gdf = DataFrame.from_pandas(pdf)
```

Comment:
Wouldn't checking a dataframe/series with categorical index without being build with `from_pandas` be a good idea as well?

Reply:
Good point, will extend the tests.